impeller/compiler/compiler_test.cc

```diff
@@ -14,6 +15,16 @@ static fml::UniqueFD CreateIntermediatesDirectory() {
   auto test_name = flutter::testing::GetCurrentTestName();
   std::replace(test_name.begin(), test_name.end(), '/', '_');
   std::replace(test_name.begin(), test_name.end(), '.', '_');
+
+  // To be able to run tests in parallel, we can't have tests all writing to
+  // the same directory. We add the unix timestamp to the directory name.
+  // See https://github.com/flutter/flutter/issues/143330.
+  std::stringstream stream;
+  stream << test_name << "_" << std::time(nullptr);
```

Comment:
The result of std::time is implementation dependent but just returns seconds since epoch if memory serves. Resolution of a second if far too low for tests that can be launched repeatedly. I'd suggest a high resolution clock but depending on scheduling, you are just making it harder but not impossible. I believe only mkdtemp and friends I described in https://github.com/flutter/flutter/issues/143379#issuecomment-1958877687 is the safe way to go.